internal/kafka/internal/services/data_plane_kafka_test.go

```diff
@@ -848,3 +848,130 @@ func Test_DataPlaneKafkaStatus_getStatus(t *testing.T) {
+							RoutesCreated: true,
+						}, nil
+					},
+					UpdatesFunc: func(kafkaRequest *dbapi.KafkaRequest, values map[string]interface{}) *errors.ServiceError {
+						return nil
+					},
+				},
+				kafkaConfig: &defaultKafkaConf,
+			},
+			args: args{
+				kafka: &dbapi.KafkaRequest{
+					Status:    "provisioning",
+					ClusterID: "test-cluster-id",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "should not return error if kafka is not in provisioning status",
+			fields: fields{
+				clusterService: &ClusterServiceMock{
+					FindClusterByIDFunc: func(clusterID string) (*api.Cluster, *errors.ServiceError) {
+						return &api.Cluster{}, nil
+					},
+				},
+				kafkaService: &KafkaServiceMock{
+					GetByIdFunc: func(id string) (*dbapi.KafkaRequest, *errors.ServiceError) {
+						return &dbapi.KafkaRequest{
+							ClusterID:     "test-cluster-id",
+							Status:        constants2.KafkaRequestStatusProvisioning.String(),
+							Routes:        []byte("[{'domain':'test.example.com', 'router':'test.example.com'}]"),
+							RoutesCreated: true,
+						}, nil
+					},
+					UpdatesFunc: func(kafkaRequest *dbapi.KafkaRequest, values map[string]interface{}) *errors.ServiceError {
+						return nil
+					},
+				},
+				kafkaConfig: &defaultKafkaConf,
+			},
+			args: args{
+				kafka: &dbapi.KafkaRequest{
+					Status:    "ready",
+					ClusterID: "test-cluster-id",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "should return error if updateFunc returns error",
+			fields: fields{
+				clusterService: &ClusterServiceMock{
+					FindClusterByIDFunc: func(clusterID string) (*api.Cluster, *errors.ServiceError) {
+						return &api.Cluster{}, nil
+					},
+				},
+				kafkaService: &KafkaServiceMock{
+					GetByIdFunc: func(id string) (*dbapi.KafkaRequest, *errors.ServiceError) {
+						return &dbapi.KafkaRequest{
+							ClusterID:     "test-cluster-id",
+							Status:        constants2.KafkaRequestStatusProvisioning.String(),
+							Routes:        []byte("[{'domain':'test.example.com', 'router':'test.example.com'}]"),
+							RoutesCreated: true,
+						}, nil
+					},
+					UpdatesFunc: func(kafkaRequest *dbapi.KafkaRequest, values map[string]interface{}) *errors.ServiceError {
+						return errors.GeneralError("test")
+					},
+				},
+				kafkaConfig: &defaultKafkaConf,
+			},
+			args: args{
+				kafka: &dbapi.KafkaRequest{
+					Status:    "provisioning",
+					ClusterID: "test-cluster-id",
+				},
+			},
+			wantErr: true,
+		},
+	}
+
+	g := NewWithT(t)
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			d := &dataPlaneKafkaService{
+				kafkaService:   tt.fields.kafkaService,
+				clusterService: tt.fields.clusterService,
+				kafkaConfig:    tt.fields.kafkaConfig,
+			}
+			err := d.reassignKafkaClusterFull(tt.args.kafka)
+			g.Expect(err != nil).To(Equal(tt.wantErr))
```

Comment:
There is no check about the fact that the received error is exactly the error we are expecting: everything here could fail and the test would success.
I'd suggest:

1. Add a check to verify that the received error is a given error
2. Add a check that `FindClusterByIDFunc `, `GetByIdFunc` and `UpdatesFunc` are called exactly 1 time (that can be done with: `Expect(tt.fields.clusterService.calls).To(HaveLen(1))`
3. Add a check to verify that the parameters received by the mocked functions are the parameter you are expecting to receive (those parameters are inside the `tt.fields.clusterService.calls[x]` object, where `x` is the call index)

Reply:
You are right, thanks for the feedback @ziccardi :)

Reply:
Hey @ziccardi 
I went a bit overboard with the unit tests fields in the first commit, I've removed these fields again. I'm not sure if 2) & 3) can still be applied as kafkaService wont be called in my second test.

Reply:
Agree. Great job!